bot/extensions/clashofcode/views.py

```diff
@@ -0,0 +1,160 @@
+import asyncio
+import logging
+import time
+
+import discord
+from codingame.http import HTTPError
+from discord import ui
+
+from bot import core
+from bot.config import settings
+from bot.extensions.clashofcode.utils import coc_client, coc_helper, languages, modes
+
+
+def em(mode, players):
+    embed = discord.Embed(title="**Clash started**")
+    embed.add_field(name="Mode", value=mode, inline=False)
+    embed.add_field(name="Players", value=players)
+    return embed
+
+
+class CreateCocView(ui.View):
+    LANGUAGES_CUSTOM_ID = "extensions:clashofcode:languages"
+    MODES_CUSTOM_ID = "extensions:clashofcode:modes"
+    SELECT_ALL_CUSTOM_ID = "extensions:clashofcode:select_all"
+    DESELECT_ALL_CUSTOM_ID = "extensions:clashofcode:deselect_all"
+    CREATE_CUSTOM_ID = "extensions:clashofcode:create"
+
+    @discord.ui.select(
+        custom_id=LANGUAGES_CUSTOM_ID,
+        placeholder="➕ Select a language",
+        options=[discord.SelectOption(label=lang, default=True) for lang in languages],
+        max_values=len(languages),
+    )
+    async def select_languages(self, interaction: core.InteractionType, _select: ui.Select):
+        await interaction.response.defer()
+
+    @discord.ui.button(label="Select All", style=discord.ButtonStyle.blurple, emoji="🟢", custom_id=SELECT_ALL_CUSTOM_ID)
+    async def select_all_languages(self, interaction: core.InteractionType, _button: ui.Button):
+        modes_select = discord.utils.get(self.children, custom_id=self.MODES_CUSTOM_ID)
+        for option in modes_select.options:
+            option.default = not modes_select.values or option.label in modes_select.values
+
+        await interaction.response.edit_message(view=self)
+
+        for option in modes_select.options:
+            option.default = True
+
+    @discord.ui.button(
+        label="Deselect All", style=discord.ButtonStyle.blurple, emoji="🔴", custom_id=DESELECT_ALL_CUSTOM_ID
+    )
+    async def deselect_all_languages(self, interaction: core.InteractionType, _button: ui.Button):
```

Comment:
`deselect_all` ?

Reply:
we deselect all languages, so i find it helps with readability to include it in the function name.